### DISCUSSION TITLE
SWR support instead of react query

### DISCUSSION THREAD

**Discussion by shunkakinoki** [trpc/trpc#1014] [Ideas]
As a SWR user, it would be great if trpc supported SWR too, perhaps another package? (happy to work on this)

**Comment by KATT** [comment]
I won't work on this myself, there's a lot of reasons why `react-query` is a better fit for tRPC, but feel free to work on it.

**Reply by unframework** [reply]
Thanks for responding @KATT, any brief pointers on what makes SWR a worse fit? I would guess something around the calling convention and how SWR treats everything as a cache key maybe? Great work on the library, by the way.

**Reply by KATT** [reply]
No mutation support, worse ssr hydration/dehydration, waay fewer features, non-primitive input arguments leading to infinite loops when not `useMemo`'d, inferior pagination support, inferior documentation. List goes on - originally I tried using swr for trpc, but it was just a worse DX which didn't feel "complete". React query is simply superior in any angle I can see. 

I don't see any case _against_ react-query apart from saving a few bytes that no end-user would ever notice. 

It'd be easy to make an swr wrapper around `@trpc/client` and I'd happily promote it if someone made a good library, but I won't spend time building something that is a different flavour of the same thing.

**Reply by unframework** [reply]
Fair enough (and for the record, some projects do end up choosing SWR for a lot of the above reasons such as simpler docs/features haha!). And this sounds like there is no engineering showstopper preventing an SWR wrapper implementation then.

Will be watching this space very closely, thanks! Consuming GraphQL has... underdelivered on the DX front, so this is very exciting to see!

**Reply by KATT** [reply]
There is no engineering showstopper for swr in the same way there's no engineering showstopper for a Svelte, Vue, Angular, or NativeScript port ☺

_Albeit, swr would be a lot easier than the others to implement_

**Comment by sachinraja** [comment]
I've built an integration: https://github.com/sachinraja/trpc-swr
